base/uk.ac.stfc.isis.ibex.opis/resources/CCD100/updateControlsForNumberOfChannels.py

```diff
@@ -0,0 +1,33 @@
+from org.csstudio.opibuilder.scriptUtil import PVUtil
+from org.csstudio.opibuilder.scriptUtil import ConsoleUtil
+from time import sleep
+
+def is_active_macro(value):
+    if value is None or len(value)==0:
+        return False  # Empty macro
+    elif len(value)>=3 and value[:2]=="$(" and value[-1]==")":
+        return False  # Unexpanded macro
+    else:
+        return True
+    
+def get_count():
+    count = 0
+    macros = display.getPropertyValue("macros").getMacrosMap()
+    for i in range(1,5):
+        if is_active_macro(macros.get("CCD_"+str(i))):
+            count = i
```

Comment:
This variable has a misleading name. If I define only `CCD_1` and `CCD_3`, a variable named "count" implies that it's value should be 2, but it is actually 3.

Reply:
Currently the macros have to be contiguous. Maybe I can make it so that doesn't have to be the case...

Reply:
... Done